include/envoy/network/address.h

```diff
@@ -104,6 +104,7 @@ class Instance {
   virtual ~Instance() = default;
 
   virtual bool operator==(const Instance& rhs) const PURE;
+  virtual bool equalsExceptPort(const Instance& rhs) const PURE;
```

Comment:
IMO this is an abstraction leak to expose port up to this level. I would remove this interface change.